llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -928,6 +928,19 @@ InstructionCost VPInstruction::computeCost(ElementCount VF,
     return Ctx.TTI.getVectorInstrCost(Instruction::ExtractElement, VecTy,
                                       Ctx.CostKind);
   }
+  case Instruction::Select: {
+    if (!getUnderlyingValue())
```

Comment:
```suggestion
    if (!isLiveIn() || !getUnderlyingValue())
```

Reply:
Is it possible for a VPInstruction to be a live in? A live in has to not have a recipe:

```c++
  /// Returns true if this VPValue is a live-in, i.e. defined outside the VPlan.
  bool isLiveIn() const { return !hasDefiningRecipe(); }
```